oeps/oep-0037/decisions/0001-yaml-centralization.rst

```diff
@@ -0,0 +1,34 @@
+Centralization of Development Data Yaml files
+=============================================
+
+Status
+------
+
+Draft
+
+Context
+-------
+
+Devstack currently has no standard way of creating dev data. Overtime, developers have figured out their own ways to create data for local development. Recently, `OEP-37`_ was proposed, which suggests a standardized method for creating dev data. We decided the proposal was good enough to warrant a prototype and during the prototyping phase, we ran into the question of how much of this method we want to centralize into devstack, vs keep things decentralized by keeping as much as is possible in each IDAs.
+
+Specifically, this ADR pertains to question of where to put yaml files specifiying what data to load into the devstack development environment. As it is implemented, each IDA will have its own yaml files which lay out what data to create for that IDA.
+
+
+Decision
+--------
+
+Since this method is meant to be used in devstack to orchestrate dev data across multiple IDAs, we decided to put all of the data yaml files into the devstack repo. This decision is especially relavant for yaml files specifiying what data to create during provisioning and for very specific common workflows. The hope is that developers will only have to modify yaml files in devstack repo to create more custom data.
```

Comment:
[suggestion] Probably want to include something about the following issues with localizing the data to the IDA repos:

- Having mismatched devstack and IDA repos checked out could result in problems parsing and loading the data (including lots of irritating debugging sessions) when devstack's loader doesn't match an IDA's data or two IDAs have incompatible data
- ...and I think there was another reason, but I can't remember it. :sweat_smile:

Reply:
Clarification: devstack's loader really only passes the yaml files to `load_dev_data` management commands in each IDA. So this bullet point doesn't completely apply.

For reference,[ line which passes yaml file to management command](https://github.com/edx/devstack/blob/9fed63d386fedbe5da897e82cd66325ea1e12598/load_dev_data.py#L23)

Reply:
Ah, OK. So it's really just that two IDAs might be out of sync with each other. (Or devstack might refer to a file that doesn't exist yet in the checked out version of the IDA, I suppose.)

Reply:
The worry of two repos being out of sync with each other is true for this project and in devstack usage in general. Here, the extra sync problem would arise if the yaml file in devstack has a model name that is not yet supported in user's checked out version of a IDA.

Reply:
Makes sense.